sql/analyzer/fix_field_indexes.go

```diff
@@ -18,28 +18,42 @@ import (
 	"github.com/dolthub/go-mysql-server/sql"
 	"github.com/dolthub/go-mysql-server/sql/expression"
 	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/transform"
 )
 
 // FixFieldIndexesOnExpressions executes FixFieldIndexes on a list of exprs.
-func FixFieldIndexesOnExpressions(ctx *sql.Context, scope *Scope, a *Analyzer, schema sql.Schema, expressions ...sql.Expression) ([]sql.Expression, error) {
-	var result = make([]sql.Expression, len(expressions))
-	for i, e := range expressions {
-		var err error
-		result[i], err = FixFieldIndexes(ctx, scope, a, schema, e)
+func FixFieldIndexesOnExpressions(ctx *sql.Context, scope *Scope, a *Analyzer, schema sql.Schema, expressions ...sql.Expression) ([]sql.Expression, transform.TreeIdentity, error) {
+	var result []sql.Expression
+	var res sql.Expression
+	var same transform.TreeIdentity
+	var err error
+	for i := 0; i < len(expressions); i++ {
```

Comment:
i've been using for loops for hot code paths because it's about ~10% faster. I'm not sure whether this qualifies as a hot path? But it's the same algorithm as the others, for consistency

Reply:
TIL difference between one and two variable range statements